contracts/Org.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+contract Org {
```

Comment:
The design here is to have a separate contract instance per org. It's rather expensive and non-trivial to use. It's also impossible to introduce any updating mechanism, the contract is ultimately set in stone. Would it be possible to have a single instance of the contract? Is having a fully customized instance our desired use case?

Reply:
I *think* this is the right design.

* For upgrades, it's simply a matter of pointing the Org's ENS name to a new contract. The big difference is that it's not a global upgrade of all org contracts. This has pros and cons.
* Having a contract per org gives orgs extra flexibility. It's theoretically possible to publish a different contract per org with the same ABI.
* Orgs having their own accounts is, I think, quite important. I'm a little skeptical of a single contract managing the money of *all* orgs for example. I would be worried about this from a security point of view.
* I don't really see it being expensive: yes, the initial deployment is more expensive, but once that's out of the way, it should be slightly cheaper to do day-to-day operations (eg. anchoring), since the contract is simpler. This should amortize over time :thinking: 
* The non-trivial aspect I guess is that users need to deploy this contract. However they won't have to do that manually: this is something that we should handle in an "OrgFactory" contract, that deploys the "Org" contract for you. This gives us a bunch of flexibility, and allows us to change the default org contract that is deployed for *new* deployments, as well as charge or reimburse the user as needed. This pattern is very common with multi-sig wallets.
* On the other hand, the Org contract is easier to audit and get right with this design, then an "OrgRegistry" contract that has to handle all orgs.

It's hard to know exactly if this decision will be the right one today, but I tend to prefer this model to the "global" contract model for the reasons above.

Reply:
I can't argue with any of that, it makes perfect sense.

So most importantly we're providing an API, the implementation is just a help for the casual, lazy users. I'm not sure what's the convention, should we publish a separate `.sol` file with just the interface, so others can easily import it and then implement or use it?

Reply:
Yeah that's a good point. I think in the beginning, 99% of users will just use this contract, but eventually we might want to standardize on an interface. I would say let's open an issue for that. I don't want to prematurely define the interface, but I think we should have it set in stone for the official launch.

Reply:
#41